src/main/java/org/jenkinsci/plugins/saml/BundleKeyStore.java

```diff
@@ -0,0 +1,277 @@
+package org.jenkinsci.plugins.saml;
+
+import sun.security.x509.AlgorithmId;
+import sun.security.x509.CertificateAlgorithmId;
+import sun.security.x509.CertificateSerialNumber;
+import sun.security.x509.CertificateValidity;
+import sun.security.x509.CertificateVersion;
+import sun.security.x509.CertificateX509Key;
+import sun.security.x509.X500Name;
+import sun.security.x509.X509CertImpl;
+import sun.security.x509.X509CertInfo;
```

Comment:
bouncycastle could work the method toCertificate return the Certificate that I need but I never used this library and I do not know how maintained it is, this certificates classes are used by keytool and it is difficult to think that could be removed, I am gonna try to use the bouncycastle library.